lib/calendar/recurring.js

```diff
@@ -50,6 +50,27 @@ const fillOccurrencesBetween = (start, end, iterator, eventDuration, internalDts
     return result;
 };
 
+/**
+ * Convert the until property of an rrule to be in the timezone of the start date
+ */
+const getModifiedUntilRrule = (internalRrule, startTzid) => {
+    if (!internalRrule || !internalRrule.value || !internalRrule.value.until || !startTzid) {
+        return internalRrule;
+    }
+    const utcUntil = toUTCDate(internalRrule.value.until);
```

Comment:
`value.until` will not be a UTC date for all-day events. Will that pose any problem?